lib/docker-utils.coffee

```diff
@@ -302,7 +302,7 @@ class RdtDockerUtils
 			containerId = containerInfo.Id
 
 			Promise.try ->
-				if semver.lt(dockerVersion, '1.10.0', true)
+				if rSemver.lt(dockerVersion, '1.10.0')
```

Comment:
This isn't safe. That `, true` is there to enable slightly more relaxed semver handling (it allows 0-prefixed version parts), because Docker moved this year from versions like `1.10.0` to Ubuntu style versions like `17.06.0-ce` (June 2017, Community Edition). Right now, this line (correctly) returns `false` for those new versions, with this change it'll return `true`, and break things for some of the upcoming ResinOS versions.

This is a pretty hacky fix for this though (and I must admit, I wrote it), since we're pretending that 17.06.0 is a semver version, which it really isn't.

I think we can handle this better and correctly using `resin-semver` with logic more like:

```coffeescript
if rSemver.isValid(version) && rSemver.lt(version, '1.10.0'))
  return containerId

# Else: either it's a release after 1.10.0, or its one of the fun new non-semver versions,
# which we incidentally know all appeared after 1.10.0
```

Reply:
Oh, also whatever we do here should probably also be done in https://github.com/resin-io-modules/docker-toolbelt/blob/master/lib/docker-toolbelt.coffee#L81, which currently duplicates the exact same logic.

Reply:
Yep yep this sounds good. I've added docker-toolbelt to the HQ issue as well